operator/internal/central/extensions/reconcile_scanner_v4_feature_defaults.go

```diff
@@ -0,0 +1,84 @@
+package extensions
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/operator-framework/helm-operator-plugins/pkg/extensions"
+	"github.com/pkg/errors"
+	platform "github.com/stackrox/rox/operator/api/v1alpha1"
+	"github.com/stackrox/rox/operator/internal/common/defaulting"
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	annotationKey = defaulting.FeatureDefaultKeyScannerV4
+)
+
+// This extension's purpose is to
+//
+//   1. apply defaults by mutating the Central spec as a prerequisite for the value translator
+//   2. persist any implicit Scanner V4 Enabled|Disabled setting in the Central annotations for later usage during upgrade-reconcilliations.
+//
+
+func ReconcileScannerV4FeatureDefaultsExtension(client ctrlClient.Client) extensions.ReconcileExtension {
+	return wrapExtension(reconcileScannerV4FeatureDefaults, client, nil)
+}
+
+func reconcileScannerV4FeatureDefaults(
+	ctx context.Context, central *platform.Central, client ctrlClient.Client, _ ctrlClient.Reader,
+	_ func(updateStatusFunc), logger logr.Logger) error {
+	scannerV4Spec := initializedDeepCopy(central.Spec.ScannerV4)
+
+	var scannerComp platform.ScannerV4ComponentPolicy
+	if scannerV4Spec.ScannerComponent != nil {
+		scannerComp = *scannerV4Spec.ScannerComponent
+	}
+
+	if scannerComp == platform.ScannerV4ComponentEnabled || scannerComp == platform.ScannerV4ComponentDisabled {
+		// User provided an explicit choice, nothing to do in this extension.
+		return nil
+	}
+
+	// User is relying on defaults. Compute default and persist corresponding annotation.
+
+	componentPolicy := defaulting.ScannerV4ComponentPolicy(logger, &central.Status, central.GetAnnotations(), scannerV4Spec)
+	scannerV4Spec.ScannerComponent = &componentPolicy
+
+	if central.Annotations == nil {
+		central.Annotations = make(map[string]string)
+	}
+	if central.Annotations[annotationKey] == "" {
```

Comment:
Shouldn't we have a more general check? 🤔 
```suggestion
	if central.Annotations[annotationKey] != string(componentPolicy) {
```

Reply:
Hah. Until 1h ago this was the logic I had in the PR. But then I added [another test](https://github.com/stackrox/stackrox/blob/828bd4edec9f1ccd780c4dd5d081a792d5a8c03e/operator/internal/central/extensions/reconcile_scanner_v4_feature_defaults_test.go#L99) and changed it to `== ""` because I thought, when I don't understand what the current value of the annotation means, I better stay away from it.

That being said, I am still not 100% convinced either way.

Reply:
I'm fine with it.

See commits 34f2c49b82ca5fb6a4ccf2a0b7801eee7f7a885f..740b15fd8faedc1df945fbe7d50a05888f124481. 
Unfortunately there were problems with the unit tests, which required me to hack a dirty workaround. :-(

Reply:
ok to close?